WcaOnRails/app/models/competition.rb

```diff
@@ -466,22 +466,76 @@ def country
   end
 
   def self.search(query, params: {})
-    sql_query = "%#{query}%"
-    Competition.where("id LIKE :sql_query OR name LIKE :sql_query OR cellName LIKE :sql_query OR cityName LIKE :sql_query OR countryId LIKE :sql_query", sql_query: sql_query).order(year: :desc, month: :desc, day: :desc)
+    competitions = Competition.where(showAtAll: true)
```

Comment:
Are you aware that each `.where` call actually executes the SQL query?

I think you want to build the where and order clauses separately, and only at the end do `Competition.where(where_clause).order(order_clause)`.